### ISSUE TITLE
Raise Exception when `GammaGammaModel` does not converge properly

### ISSUE THREAD

**Issue by ColtAllen** [pymc-labs/pymc-marketing#594]
If a fitted `GammaGammaModel` has a  `q` parameter less than 1, all predicted spend values will be negative. This usually happens when the model is fit to heavy-tailed monetary value arrays with values in the millions.

We should add a check for this raising the following exception:

`ValueError: Model did not converge properly. Scale monetary_value array by maximum value and refit model.`

**Comment by ricardoV94** [comment]
I don't think we should raise after fitting. It's fine to have `q < 1`, except some expectation methods fail then (but not distributional methods for instance). We should warn/raise only on the specific methods where `q < 1` is problematic.
